tcf_website/management/commands/load_grades.py

```diff
@@ -0,0 +1,398 @@
+        )
+    '''
+
+    def handle(self, *args, **options):
+
+        # self.verbose = options['verbose']
+
+        self.data_dir = 'tcf_website/management/commands/grade_data/excel/'
+
+        # semester = options['semester']
+
+        [self.load_semester_file(file) for file in sorted(
+            os.listdir(self.data_dir)) if not file.startswith('.')]
+
+        '''
+        if semester == 'ALL_DANGEROUS':
+            # ignores temporary files
+            [self.load_semester_file(file) for file in sorted(os.listdir(self.data_dir)) if not file.startswith('.')]
+        else:
+            self.load_semester_file(f"{semester.lower()}.xlsx")
+        '''
+
+    def clean(self, df):
+        CourseGrade.objects.all().delete()  # deletes CourseGrade table
+        # deletes CourseInstructorGrade table
+        CourseInstructorGrade.objects.all().delete()
+        return df.dropna(
+            how="all",
+            subset=[
+                'A+',
+                'A',
+                'A-',
+                'B+',
+                'B',
+                'B-',
+                'C+',
+                'C',
+                'C-',
+                'D+',
+                'D',
+                'D-',
+                'F'])
+
+    def load_semester_file(self, file):
+        year, semester = file.split('.')[0].split('_')
+        year = int(year)
+        season = semester.upper()
+
+        print(year, season)
+
+        df = self.clean(pd.read_excel(os.path.join(self.data_dir, file)))
+
+        print(f"{df.size} sections")
+
+        for index, row in tqdm(df.iterrows(), total=df.shape[0]):
+            # print(str(row).encode('ascii', 'ignore').decode('ascii'))
+            self.load_row_into_dict(row)
+            # break
+
+        self.load_dict_into_models()
+
+    def load_row_into_dict(self, row):
+        try:
+            # parsing fields of excel
+
+            first_name = row['Instructor First Name']
+            middle_name = row['Instructor Middle Name']
+            last_name = row['Instructor Last Name']
+            email = row['Instructor Email']
+            subdepartment = row['Subject']
+            number = re.sub('[^0-9]', '', str(row['Course Number']))
+            section_number = row['Section Number']  # not used
+            title = row['Title']
+            gpa = float(row['Course GPA'])
+            a_plus = int(row['A+'])
+            a = int(row['A'])
+            a_minus = int(row['A-'])
+            b_plus = int(row['B+'])
+            b = int(row['B'])
+            b_minus = int(row['B-'])
+            c_plus = int(row['C+'])
+            c = int(row['C'])
+            c_minus = int(row['C-'])
+            d_plus = int(row['D+'])
+            d = int(row['D'])
+            d_minus = int(row['D-'])
+            f = int(row['F'])
+            ot = int(row['OT'])
+            drop = int(row['DR'])
+            withdraw = int(row['W'])
+            # credit
+            # general_credit
+            # no_credit
```

Comment:
Seems like these 3 fields were originally planned to be part of the model, but they decided against it — should remove these comments at some point

Reply:
Removed in c45f4454ed25cb07eb138184b1b935b9661baefb